File: solutions/cpp/darts/1/darts.cpp
#include "darts.h"

int
darts::score(const double& x, const double& y) {
    double rad = std::sqrt(std::pow(x, 2) + std::pow(y, 2));
    if(rad <= 1)
        return 10;
    if(rad <= 5)
        return 5;
    if(rad <= 10)
        return 1;
    return 0;
}

File: solutions/java/darts/1/src/main/java/Darts.java
class Darts {

    float distance;
	Darts(double x, double y) {
		distance = (float)Math.sqrt( Math.pow( x, 2 ) + Math.pow( y, 2 ) );
    }

    int score() {
		if( distance <= 1 )
			return 10;
		else if( distance <= 5 )
			return 5;
		else if( distance <= 10 )
			return 1;
		return 0;
    }

}
